Set DeploymentReady to False when ReadyCount will be 0

mrkisaolamb opened the pull request:
Make sure that DeploymentReady condition is set back to False if the Deployment ReadyReplicas falls back to 0

SeanMooney reviewed (approved):
ah this is the same int to time.duration change looks ok to me

bogdando reviewed (commented):
lgtm

abays commented on controllers/placementapi_controller.go:
```diff
@@ -766,7 +766,17 @@ func (r *PlacementAPIReconciler) reconcileNormal(ctx context.Context, instance *
 
 	if instance.Status.ReadyCount > 0 {
 		instance.Status.Conditions.MarkTrue(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage)
+	} else {
+		Log.Info("Deployment is not ready")
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.DeploymentReadyCondition,
+			condition.RequestedReason,
+			condition.SeverityInfo,
+			condition.DeploymentReadyRunningMessage))
+		// It is OK to return success as we are watching for StatefulSet changes
+		return ctrl.Result{}, nil
```
What if `replicas` is set to `0`?  That might prevent the CR from ever reaching the "ready" state [1].

[1] https://github.com/openstack-k8s-operators/placement-operator/blob/4daf37ecb2bb584e1f3a98b7f1660b81519f8ff1/api/v1beta1/placementapi_types.go#L196-L199

gibizer replied:
@abays you are right. If the replicas is 0 then we can be ready with 0 running pods.

mrkisaolamb replied:
done

mrkisaolamb committed:
Set DeploymentReady to False when ReadyCount will be 0

Make sure that DeploymentReady condition is set back to False
if the Deployment ReadyReplicas falls back to 0

bogdando commented:
/LGTM
